src/compiler/program.ts

```diff
@@ -2722,6 +2721,18 @@ namespace ts {
             return result;
         }
 
+        function getCreateSourceFileOptions(fileName: string, moduleResolutionCache: ModuleResolutionCache | undefined, host: CompilerHost, options: CompilerOptions) {
+            // It's a _little odd_ that we can't set `impliedNodeFormat` until the program step - but it's the first and only time we have a resolution cache

```

Comment:
Actually, is this comment still relevant after being moved? We don’t even have a SourceFile yet.

Reply:
We call this to make the options to make a source file, so I feel it's still correct.